src/app/app.module.ts

```diff
@@ -1,18 +1,40 @@
 import { NgModule } from '@angular/core';
+import { NgbModule } from '@ng-bootstrap/ng-bootstrap';
+
+import { HttpClientModule, HTTP_INTERCEPTORS } from '@angular/common/http';
 import { BrowserModule } from '@angular/platform-browser';
 
 import { AppRoutingModule } from './app-routing.module';
 import { AppComponent } from './app.component';
 
+import { ErrorInterceptor } from './core/interceptors/error.interceptor';
+import { RetryInterceptor } from './core/interceptors/retry.interceptor';
+import { LoaderComponent } from './shared/components/loader/loader.component';
+import { HeaderComponent } from './shared/components/header/header.component';
+import { FooterComponent } from './shared/components/footer/footer.component';
+import { PageNotFoundComponent } from './core/components/page-not-found/page-not-found.component';
+
 @NgModule({
   declarations: [
-    AppComponent
+    AppComponent,
+    LoaderComponent,
+    HeaderComponent,
+    FooterComponent,
+    PageNotFoundComponent,
   ],
-  imports: [
-    BrowserModule,
-    AppRoutingModule
+  imports: [BrowserModule, NgbModule, HttpClientModule, AppRoutingModule],
+  providers: [
+    {
+      provide: HTTP_INTERCEPTORS,
+      useClass: RetryInterceptor,
+      multi: true,
+    },
+    {
+      provide: HTTP_INTERCEPTORS,
+      useClass: ErrorInterceptor,
+      multi: true,
+    },
```

Comment:
You have loader interceptor added to project but not here in providers

Reply:
Sure, fixed: 42fdd5a1a4a206f33cfd5df4299d474ac537d450